back/app/services/permissions/idea_permissions_service.rb

```diff
@@ -12,10 +14,18 @@ def initialize(idea, user, user_requirements_service: nil)
     def denied_reason_for_action(action, reaction_mode: nil)
       reason = super
       return reason if reason
+      return if user && UserRoleService.new.can_moderate_project?(idea.project, user)
```

Comment:
I'm a little unsure about this, as the amount of queries is very context dependent, so all cases can't be covered by these tests very well, and it's also not the only factor for performance.

Feels like that in 90+% of serialized ideas, the user is not the author nor an admin/moderator, it feels like a simple and effective optimization to have.

If you're confident that it follows a very efficient codepath in the most common case of a serialized idea (I'm a resident, the input is someone else's input) than I'm fine with it.

I don't immediately understand where this condition (the user is not the author) is checked right now?

Reply:
@kogre At the moment the condition is checked in the idea policy (update?). But to optimize the action descriptors, we would have to check for the author in the permissions services, where the only way to deny the action, is to provide a denied reason. If we do this, I'm not sure if the frontend will still work well.

Actually, I don't think that the impact on performance is acceptable, mostly because of the relationship with reactions that is used. But perhaps the `vbii` precalculated values passed from the controller to the serializer can be used instead.

In either case, this doesn't feel like something I would be able to wrap up today. I've been struggling the whole day with getting the cleanup PR green.